app/scripts/models/account.js

```diff
@@ -298,24 +312,37 @@ define([
           return fxaClient.changePassword(email, oldPassword, newPassword);
         })
         .then(function () {
-          // sign the user in, keeping the current sessionTokenContext. This
-          // prevents sync users from seeing the `sign out` button on the
-          // settings screen.
           return fxaClient.signIn(
             email,
             newPassword,
             relier,
             {
-              reason: fxaClient.SIGNIN_REASON.PASSWORD_CHANGE,
-              sessionTokenContext: self.get('sessionTokenContext')
+              reason: fxaClient.SIGNIN_REASON.PASSWORD_CHANGE
```

Comment:
Tangent: maybe we should create new methods like `fxaClient.signInAfterPasswordChange` to keep these "reasons" encapsulated within fxaClient.